test/vc/VVVVCTestBase.sol

```diff
@@ -311,4 +335,225 @@ abstract contract VVVVCTestBase is Test {
+        address[] memory _projectTokenProxyWallets,
+        uint256[] memory _investmentRoundIds,
+        uint256 _tokenAmountToClaim,
+        uint256[] memory _investedPerRound,
+        bytes32[] memory _investmentLeaves,
+        bytes32[][] memory _investmentProofs
+    ) public view returns (VVVVCAlternateTokenDistributor.ClaimParams memory) {
+        VVVVCAlternateTokenDistributor.ClaimParams memory params = VVVVCAlternateTokenDistributor
+            .ClaimParams({
+                callerAddress: _callerAddress,
+                userKycAddress: _kycAddress,
+                projectTokenAddress: address(ProjectTokenInstance),
+                projectTokenProxyWallets: _projectTokenProxyWallets,
+                investmentRoundIds: _investmentRoundIds,
+                tokenAmountToClaim: _tokenAmountToClaim,
+                deadline: block.timestamp + 1 hours,
+                signature: bytes("placeholder"),
+                investedPerRound: _investedPerRound,
+                investmentLeaves: _investmentLeaves,
+                investmentProofs: _investmentProofs
+            });
+
+        bytes memory sig = getEIP712SignatureForAlternateClaim(
+            alternateTokenDistributorDomainSeparator,
+            alternateClaimTypehash,
+            params
+        );
+
+        params.signature = sig;
+
+        return params;
+    }
+
+    //functions to prepare token claim params and avoid duplicating code. for a set of investment rounds: creates trees, sets roots on read-only ledger, creates merkle proofs for the given user indices (position in that investment round's array of investor kyc addresses), creates valid claim signature for that user, and returns a ClaimParams object containing all info to validate a user's prior investment(s) and currently permitted claim(s)
+    function prepareAlternateDistributorClaimParams(
+        address _callerKycAddress
+    ) public returns (VVVVCAlternateTokenDistributor.ClaimParams memory) {
+        return prepareAlternateDistributorClaimParams(_callerKycAddress, 1);
+    }
+
+    function prepareAlternateDistributorClaimParams(
+        address _callerKycAddress,
+        uint256 _usersArrayIndex
+    ) public returns (VVVVCAlternateTokenDistributor.ClaimParams memory) {
+        uint256[] memory investedAmountsArray = new uint256[](users.length);
+        uint256[] memory placeholderArray = new uint256[](projectTokenProxyWallets.length);
+        uint256[] memory userInvestedAmounts = new uint256[](projectTokenProxyWallets.length);
+
+        //set the caller kyc address to the user array at the index of _usersArrayIndex
+        users[_usersArrayIndex] = _callerKycAddress;
+
+        AlternateDistributorInvestmentDetails memory details = AlternateDistributorInvestmentDetails({
+            investedAmounts: investedAmountsArray, //all users in users array, not just selected user
+            userIndices: placeholderArray,
+            investmentRoundIds: placeholderArray,
```

Comment:
Your glance was correct I think! Huh, so both just pointed to the array, where I was thinking it would copy the array to each struct field separately.